gadget/mkfs.go

```diff
@@ -29,6 +29,15 @@ import (
 	"github.com/snapcore/snapd/osutil"
 )
 
+type MkfsFunc func(imgFile, label, contentsRootDir string) error
+
+var (
+	mkfsHandlers = map[string]MkfsFunc{
```

Comment:
We'll need mkfsHandlers exported to be used in MakeFilesystems() in gadget/install.

Reply:
as we mentioned in the end we might need an internal package.  A possible organisation to consider could be:

```
    gadget
    gadget/internal
    MAYBE gadget/layout or gadget/volume
    gadget/update
    gadget/install
```